src/server/bg_services/log_replication_scanner.js

```diff
@@ -0,0 +1,219 @@
+
+        return true;
+    }
+
+    async scan() {
+        if (!this.noobaa_connection) throw new Error('noobaa endpoint connection is not started yet...');
+
+        const replications = await replication_store.find_aws_log_based_replication_rules();
+
+        await P.all(_.map(replications, async repl => {
+            _.map(repl.rules, async rule => {
+                const replication_id = repl._id;
+
+                const { src_bucket, dst_bucket } = this.find_src_and_dst_buckets(rule.destination_bucket, replication_id);
+                if (!src_bucket || !dst_bucket) {
+                    dbg.error('log_replication_scanner: can not find src_bucket or dst_bucket object', src_bucket, dst_bucket);
+                    return;
+                }
+
+                const candidates = await log_formatter.get_aws_log_candidates(repl.aws_log_replication_info);
+
+                const keys_sizes_map_to_copy = await this.head_buckets_and_compare(src_bucket.name, dst_bucket.name, candidates);
+
+                dbg.log1('log_replication_scanner: keys_sizes_map_to_copy: ', keys_sizes_map_to_copy);
+                let move_res;
+                const keys_to_copy = Object.keys(keys_sizes_map_to_copy);
+                if (keys_to_copy.length) {
+                    const copy_type = this._get_copy_type();
+                    move_res = await this.move_objects(copy_type, src_bucket.name, dst_bucket.name, keys_to_copy);
+                    console.log('replication_scanner: scan move_res: ', move_res);
+                }
+            });
+        }));
+    }
+
+    async head_buckets_and_compare(src_bucket, dst_bucket, candidates) {
+        const keys_sizes_map_to_copy = {};
+
+        // edge case 1: candidates = [] , nothing to replicate
+        if (!candidates.Contents.length) return keys_sizes_map_to_copy;
+
+        // head candidates in src_bucket and dst_bucket
+        const src_dst_objects_list = await this.head_objects(src_bucket, dst_bucket, candidates);
+
+        // edge case 1: src_head_map = {} , nothing to replicate
+        if (!src_dst_objects_list.length) return keys_sizes_map_to_copy;
+
+        for (let candidate of src_dst_objects_list) {
+
+            // case 1: object not found in src
+            if (!candidate.src_object_info) {
+                break;
+            }
+
+            // case 2: object not found in dst
+            if (!candidate.dst_object_info) {
+                keys_sizes_map_to_copy[candidate.key] = candidate.ContentLength;
+                break;
+            }
+
+            // case 3: object present both on the src and dst, need to check_data_or_md_change
+            if (this.check_data_or_md_changed(candidate.src_object_info, candidate.dst_object_info)) {
+                keys_sizes_map_to_copy[candidate.key] = candidate.ContentLength;
+            }
+        }
+        return keys_sizes_map_to_copy;
+    }
+
+    check_data_or_md_changed(src_info, dst_info) {
```

Comment:
all the following functions can be reused as they have almost if not identical code as in the regular replication flow.
I would recommend creating replication_utils.js file and using the same code.